pages/blog/index.tsx

```diff
@@ -0,0 +1,94 @@
+"use client";
+import { NextPage } from "next";
+import { useState, useEffect, useCallback } from "react";
+import Loader from "../../components/blog/Loader";
+import PageHeader from "../../components/blog/PageHeader";
+import PostList, { Article } from "../../components/blog/PostList";
+import Footer from "../../components/Footer";
+import { fetchAPI } from "../../utils/blog/fetch-api";
+import Navbar from "../../views/Navbar";
+
+
+interface Meta {
+	pagination: {
+		start: number;
+		limit: number;
+		total: number;
+	};
+}
+const Profile: NextPage = () => {
+	const [meta, setMeta] = useState<Meta | undefined>();
+	const [data, setData] = useState<Article[]>([]);
+	const [isLoading, setIsLoading] = useState(true);
+
+	const fetchData = useCallback(async (start: number, limit: number) => {
+		setIsLoading(true);
+		try {
+			const token = process.env.NEXT_PUBLIC_STRAPI_API_TOKEN;
+			const path = `/articles`;
+			const urlParamsObject = {
+				sort: { createdAt: "desc" },
+				populate: {
+					cover: { fields: ["url"] },
+					category: { populate: "*" },
+					authorsBio: {
+						populate: "*",
+					},
+				},
+				pagination: {
+					start: start,
+					limit: limit,
+				},
+			};
+			const options = { headers: { Authorization: `Bearer ${token}` } };
+			const responseData = await fetchAPI(path, urlParamsObject, options);
+
+			if (start === 0) {
+				setData(responseData.data);
+			} else {
+				setData((prevData: Article[] ) => [...prevData, ...responseData.data]);
```

Comment:
If the user goes back and forth, wouldn't this lead to redundancies in the `data` state?

(LOGAF: 3/10)